gaek/ndb_json.py

```diff
@@ -128,8 +127,16 @@ def default(self, obj):
 
     obj_type = type(obj)
     # NDB Models return a repr to calls from type().
-    if obj_type not in NDB_TYPE_ENCODING and hasattr(obj, '__metaclass__'):
-      obj_type = obj.__metaclass__
+    if obj_type not in NDB_TYPE_ENCODING:
+      if hasattr(obj, '__metaclass__'):
+        obj_type = obj.__metaclass__
+      else:
+        # Try to encode subclasses of types
+        for ndb_type in NDB_TYPE_ENCODING.keys():
```

Comment:
I'm trying to think if any of the types in `NDB_TYPE_ENCODING` are subclasses of one another that might lead to this returning weird results. If that is the case it could become nondeterministic if the `.keys()` method doesn't maintain the same order (just like dictionaries vs ordered dictionaries).
Thoughts?